force_wfmanager/left_side_pane/new_entity_modal.py

```diff
@@ -153,3 +190,126 @@ def get_plugin_from_factory(self, factory):
+
+        # Remove any non-visible traits
+        view_info = [trait_name for trait_name in
+                     view_info if trait_name in
+                     self.current_model.visible_traits()]
+
+        if view_info == []:
+            return False
+
+        return True
+
+    def _get_model_description_HTML(self):
+        """Format a description of the currently selected model and it's
+        parameters, using desc metadata from the traits in
+        ${model_name}_model.py"""
+
+        # A default message when no model selected
+        if self.selected_factory is None or self.current_model is None:
+            return html_string.format("No Model Selected", "", "")
+
+        model_name = self.selected_factory.get_name()
+        model_desc = self.selected_factory.get_description()
+        view_info = self.view_structure()
+
+        # Message for a model without editable traits
+        if view_info == []:
+            return html_string.format(model_name, model_desc, "")
+
+        # Remove traits in the view which are not editable
+        view_info = [trait_name for trait_name in
+                     view_info if trait_name in
+                     self.current_model.visible_traits()]
+
+        # Retrieve descriptions from trait metadata
+        for i, trait_name in enumerate(view_info):
+            trait = self.current_model.trait(trait_name)
+            trait_desc = trait.desc
+
+            if trait_desc is not None:
+                view_info[i] = [trait_name, trait_desc]
+            else:
+                view_info[i] = [trait_name, 'No Description Available']
+
+        # Format names as in the Instance Editor
+        view_info = [[name.replace('_', ' ').capitalize(), desc]
+                     for name, desc in view_info]
+
+        # Create a HTML string with all the model's parameters
+        body_str = ''.join([title_para.format(name, desc)
+                           for name, desc in view_info])
+
+        return html_string.format(model_name, model_desc, body_str)
+
+    def view_structure(self):
+        """Return a list of editable traits in the order
+        they appear in the view for the current model"""
+        #: The View of current_model
+        current_model_view = self.current_model.trait_view()
+
+        #: A List containing the Groups/Items in this View
+        main_group_contents = current_model_view.content.content
+
+        #: A List of items from our view in the order they appear in the view.
+        #: This function does not do anything clever for unusual view layouts.
+        main_group_items = _item_info_from_group(main_group_contents)
+        return main_group_items
+
+
+def _item_info_from_group(group_contents, item_info=None):
+    """Gets the item names from a list of groups (group_contents).
+    Returns a list of trait names corresponding to the items in the group.
+    """
+    if item_info is None:
+        item_info = []
+
+    for object in group_contents:
```

Comment:
use obj. object is a class name in the global scope, you don't want to overwrite global scope objects.